tests/data/Style/NumberFormat.php

```diff
@@ -416,6 +416,16 @@
         'The Boathouse',
         '"Meet me @ "@" @ 16:30"',
     ],
+    [
+        '"',
+        '"',
+        '@',
+    ],
+    [
+        '$100',
+        '$100',
+        '@',
+    ],
```

Comment:
Here are some cases which I think can be added (there is probably some overlap with yours). I might think of others, but I think these will fully exercise your code and won't change the results of 3344. If you get failures with these tests, I will see how your code might be altered appropriately. There are still probably some gaps, but we're getting into awfully contrived examples if there are, and we can deal with them later if they cause problems.
```php
    'issue 4242-0' => [
        'General $200 - 200', // expected result
        'General $200 - 200', // cell contents
        NumberFormat::FORMAT_GENERAL, // cell style
    ],
    'issue 4242-1' => [
        'Text $200 - 200',
        'Text $200 - 200',
        NumberFormat::FORMAT_TEXT,
    ],
    'issue 4242-2' => [
        '"Hello" she said and "Hello" I replied',
        '"Hello" she said and "Hello" I replied',
        NumberFormat::FORMAT_TEXT,
    ],
    'issue 4242-3' => [
        'Text: $200 - 200',
        '$200 - 200',
        '"Text: "' . NumberFormat::FORMAT_TEXT,
    ],
```

Reply:
Indeed those test cases overlap with all the situations I had defined before. Thanks for providing those.
I added the tests and fixed the code. I had to readd the dollar sign check inside the IF statement (which does make sense)

Reply:
I just noticed I overwrote your commit, I'm readding the code, sorry about that.